runtime/src/main/cpp/ObjCExport.mm

```diff
@@ -326,18 +326,21 @@ - (instancetype)copyWithZone:(NSZone *)zone {
   return typeInfo->writableInfo_->objCExport.typeAdapter;
 }
 
-static Protocol* getProtocolForInterface(const TypeInfo* interfaceInfo) {
+static void addProtocolForAdapter(Class clazz, const ObjCTypeAdapter* protocolAdapter) {
+  Protocol* protocol = objc_getProtocol(protocolAdapter->objCName);
+  if (protocol != nullptr) {
+    class_addProtocol(clazz, protocol);
+    class_addProtocol(object_getClass(clazz), protocol);
+  } else {
+    // TODO: construct the protocol in compiler instead, because this case can't be handled easily.
```

Comment:
Shalln't we assert/inform user here?

Reply:
I don't think so.
This can happen if the protocol is not present in any binary. It's possible only if the protocol isn't referenced at compile time. But operations like type checks do reference the protocol, so it's not easy to observe this case.
Btw, it shouldn't happen while we have #2818 applied, but I would prefer not to rely upon it.